src/app/core/services/phase.service.ts

```diff
@@ -97,14 +97,15 @@ export class PhaseService {
   }
 
   /**
-   * Determines the github's level of repo permission required for the phase.
-   * Ref: https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/#available-scopes
+   * Gets the session data from local storage and returns it.
    */
-  githubRepoPermissionLevel(): string {
-    if (this.sessionData.openPhases.includes(Phase.phaseModeration)) {
-      return 'repo';
+  getSessionData(): Observable<SessionData> {
+    if (localStorage.getItem('sessionData') === null) {
+      return from(localStorage.getItem('sessionData')).pipe(
```

Comment:
I think we no longer need to save the session data to local storage.
This is the current flow:
- `phaseService.storeSessionData` is called
- Within `phaseService.storeSessionData`, we save the session data to local storage, after fetching it
- `phaseService.sessionSetup` is called, and `getSessionData` gets called here
- Within `getSessionData`, we get the session data from local storage

The local storage usage is actually not needed now --> we could just pass the session data directly by setting it as a property of the PhaseService.
How about we remove the `getSessionData` method too ?

Just for reference:
Previously, we were using the local storage because:
- We fetched the session data before starting the OAuth process, and saved it to local storage
- We read the session data from local storage when GitHub redirected back to CATcher

But the previous flow was also a bit messy, since there was a bit of logic duplication.

Reply:
Ahhh I see, yea I was wondering what the local storage was for

alright, will make the changes necessary thanks Anubhav!